### ISSUE TITLE
WaitForObjectWithText does not work with multiple element matches

### ISSUE THREAD

**Issue by robert-altom** [robert-altom/test#503]
**In what language are the test written?** 
C#

**What version of Unity are you using?**
2018.4

**What version of AltUnityTester are you using?**
1.2.7

**On what device are you trying to run the test?**
Android, iOS

**The OS from where you are running the test?**
macOS

### Summary

WaitForObjectWithText does not work in cases where there are multiple elements matched by the By/value combination.

Example below: 

Driver.WaitForObjectWithText(By.NAME, "TitleElement", title);

If there are 2 elements that match the By.NAME "TitleElement", the function will not iterate through available elements, and will instead only try and wait for the first element to eventually match.

The reason for that seems to be because the text validation is not included in the initial check. The logic for this function seems to be:
1. Find element that matches the By/value
2. Check if the text matches the expected text
3. if not, wait and retry

Are there any workarounds for that? Would it be possible to use path and something like "//TitleElement[@text='title']"?

### Steps to reproduce
Try to use the .WaitForObjectWithText() that matches multiple results

### Relevant logs and/or screenshots

(Please offer logs and screenshots with the problem. Cover everything that is confidential)

---
<sub>You can find the original issue from GitLab [here](https://gitlab.com/altom/altunity/altunitytester/-/issues/397).</sub>

**Comment by robert-altom** [comment]
Hey @Szymonkaz we are going to implement @text selector to use it in the path #458 . This should be released with 1.6.2 or with 1.6.3.

**Comment by robert-altom** [comment]
Could the logic be updated to parse all matching elements and check for text, rather than relying on first found element, something like this?

Replace this:
```
try
	{
	    altElement = new AltUnityFindObject(SocketSettings, by, value, cameraBy,cameraPath, enabled).Execute();
        if (altElement.GetText().Equals(text))
	        break;
        ...
}
```

with:

```
try
    {
        var allMatchingElements = new AltUnityFindObjects(SocketSettings, by, value, cameraBy,cameraPath, enabled).Execute();
        altElement = allMatchingElements.Find(obj => obj.GetText() != null && obj.GetText() == text);
        if(altElement != null)
            break;
        ...
}
```
